src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -147,9 +183,15 @@ define(function (require, exports, module) {
                 documentStore = this.getFlux().store("document"),
                 currentDocument = documentStore.getDocument(documentID),
                 currentLayer = currentDocument.layers.byID(this.props.layer.id);
-
-            this.getFlux().actions.groups.setGroupExpansion(currentDocument, currentLayer,
-                !currentLayer.expanded, descendants);
+            
+            // Performance Hack: delay the expand action to promote browser paint event.
+            Promise
+                .bind(this)
+                .delay(10)
```

Comment:
according to my experiment, `delay(0)` and `delay(10)` both serve the same purpose: callback is executed in a new frame. There is also an API `setImmediate` which has a direct goal on solving this problem (https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate); sadly it is only available in IE/Edge